### ISSUE TITLE
Force values to use double quotes on attributes with non-templated text

### ISSUE THREAD

**Issue by PedroLamas** [eslint/eslint#13148]
**Please describe what the rule should do:**

Force values to use double quotes on attributes with non-templated text.

I've noticed some IDE's automatically add `{}` when one adds a new attribute, so some developers then add `''` to set the inner text, instead of deleting the curly brackets and using `""`.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Wrong:

```jsx
<Element attribute={'text'} />
```

Right:

```jsx
<Element attribute="text" />
```

**Why should this rule be included in ESLint (instead of a plugin)?**

I believe this would be useful as a global rule of ESLint for all JSX

**Are you willing to submit a pull request to implement this rule?**

I haven't yet looked into this but I can try to do so if this proposition is found to be of value!

**Comment by mdjermanovic** [comment]
Hi @PedroLamas, thanks for the new rule proposal!

This is certainly valid rule, but we don't have many JSX-specific rules in the core and this one probably wouldn't be so valuable since it already exists in `eslint-plugin-react`: [react/jsx-curly-brace-presence](https://github.com/yannickcr/eslint-plugin-react/blob/HEAD/docs/rules/jsx-curly-brace-presence.md).

For example:

```js
/* eslint react/jsx-curly-brace-presence: ["error"] */

<Element attribute={'text'} />
```

will be auto-fixed to:

```js
/* eslint react/jsx-curly-brace-presence: ["error"] */

<Element attribute="text" />
```

You can also use the core [jsx-quotes](https://eslint.org/docs/rules/jsx-quotes) rule to enforce double quotes.

**Comment by PedroLamas** [comment]
Hi @mdjermanovic very good and valid point, thanks for pointing me in the right direction!

Given that I am closing this now! 🙂
